tests/unit/test_application.py

```diff
@@ -49,7 +49,7 @@ def test_application_expand_environment(default_application, new_dir):
     project_file = Path(new_dir) / "rockcraft.yaml"
     project_file.write_text(ENVIRONMENT_YAML)
 
-    project = default_application.project
+    project = default_application.get_project("amd64", "amd64", "amd64")
```

Comment:
this part felt a bit off: it feels like if I pass the `platform` I shouldn't have to also part `build_on` and `build_for`

Reply:
Removed `build_on` as it's set to the host arch. Kept `build_for` to disambiguate between platform name and architecture name.